build/cmake/init.cmake

```diff
@@ -8,6 +8,15 @@
 # Licence:     wxWindows licence
 #############################################################################
 
+if(DEFINED wxBUILD_CXX_STANDARD AND NOT wxBUILD_CXX_STANDARD STREQUAL COMPILER_DEFAULT)
+    set(CMAKE_CXX_STANDARD ${wxBUILD_CXX_STANDARD})
+endif()
+if(CMAKE_CXX_STANDARD EQUAL 11 OR CMAKE_CXX_STANDARD EQUAL 14 OR CMAKE_CXX_STANDARD EQUAL 17)
```

Comment:
Is there really no way to say `>= 11` in CMakese?

Reply:
Yes, I tried. But then 98 is also >11.

Reply:
Oh. But it seems that 20 is also [already supported](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD), so maybe we should test for `!= 98` then?

Reply:
Yes, that is better.

I see MSVC now also supports setting the c++ version, I'll enable it for this compiler as well.

The build however fails on `std::bind2nd` because it still reports `__cplusplus 199711L` (`/Zc:__cplusplus` is not defined by CMake https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/). I'll add `wxCHECK_VISUALC_VERSION(10)` to the check.

Reply:
> The build however fails on `std::bind2nd` because it still reports `__cplusplus 199711L` 

Ugh, yes, of course, sorry for missing this.

> I'll add `wxCHECK_VISUALC_VERSION(10)` to the check.

I'm pretty sure we can use `std::bind2nd` up until MSVS 2019 if we really want to, but I don't think MSVC 10 supported range-for loops, I believe this was added in 11.

Reply:
> I'm pretty sure we can use std::bind2nd up until MSVS 2019 if we really want to, but I don't think MSVC 10 supported range-for loops, I believe this was added in 11.

Thanks. I only checked the `auto` keyword, not the range. In adition, it should also have support for lambda functions (`wxArrayString::Sort` and `wxSortedArrayString::Index`). Since the `/std:c++` switch is only available since VS2015u2, I'll be on the save side and use `wxCHECK_VISUALC_VERSION(14)`.

While looking at the `__cplusplus` usages, I also noticed that the `wxMEMBER_DELETE = delete` is not used for MSVC. It is also available since VS2015. Same for `wxUniCharRef(const wxUniCharRef&) = default`.

I'll test some more and update the PR later.